torch/fx/graph.py

```diff
@@ -64,10 +65,46 @@ def map_arg(a: Argument, fn: Callable[[Node], Argument]) -> Argument:
     else:
         return a
 
+class FinalizableList(list):
+    def __init__(self, *args, **kwargs):
+        self.finalized = False
+        super().__init__(*args, **kwargs)
+
+    mutation_methods = ['append', 'extend', 'insert', 'remove', 'pop', 'clear', 'sort', 'reverse']
+
+    def _raise_finalized(self):
+        raise RuntimeError('This Graph has already been finalized! FX graphs are immutable after '
+                           'setting output(). Please make a new graph')
+
+    def finalize(self):
+        self.finalized = True
+
+    def __deepcopy__(self, memo):
+        instance = FinalizableList()
+        instance.extend([copy.deepcopy(item, memo) for item in self])
+        if self.finalized:
+            instance.finalize()
+        return instance
+
+for method in FinalizableList.mutation_methods:
+    def scope(method):
+        def wrapper(self, *args, **kwargs):
+            if self.finalized:
+                self._raise_finalized()
+            getattr(super(FinalizableList, self), method)(*args, **kwargs)
+        return wrapper
+    setattr(FinalizableList, method, scope(method))
+
 class Graph:
-    def __init__(self):
-        self.nodes : List[Node] = []
+    def __init__(self, from_graph : Optional['Graph'] = None):
```

Comment:
It's a big odd to put this in the constructor. A graph could have a function `copy_graph` similar to copy node that would copy/append the entire graph. Then someone wanting to append to a graph can:

```
b = Graph()
a.copy_graph(b, input_fn) # same structure as copy node but copies an entire graph
```

Appending to a graph only is a specific use case, the general one is editing a graph (i.e. doing node-by-node transforms) and then splices graphs into eachother (graph_copy).